refactor SQL docs

taroface opened the pull request:
DOC-5850

Previously, subcommands for some SQL statements were documented on separate pages. This was not consistently handled, was inefficient to explain in the docs, and has fostered some deepening inconsistencies in our SQL reference documentation.

This PR updates several SQL reference docs by reuniting each statement with its subcommands on a single reference page. There should no longer be isolated pages for subcommands. Most of the affected pages in this PR have to do with updating links for the subcommand docs. 

Please **review** the following:

|                                                           Updated (please review)                                                            |                                                          Location in this PR                                                           |                                                                                          What to review                                                                                          |                                    Current version                                    |
|----------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------|
| [`ALTER DATABASE`](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-database.html)                               | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-3f61c52b06ae762b3d60c38a7be6d3eee4c1b426ec98bab4989e021686ad17ca)     | New doc organization/navigation (general comments), SQL diagram                                                                                                                                  | [Link](https://www.cockroachlabs.com/docs/stable/alter-database.html)                         |
| [`ALTER TABLE`](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-table.html)                                     | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-dcd29590f54bd74a05670346003c703a7fd7ad1f6c71cf5b2438f295f54b2e76)     | New doc organization/navigation (general comments), SQL diagram                                                                                                                                  | [Link](https://www.cockroachlabs.com/docs/stable/alter-table.html)                        |
| [`ALTER INDEX`](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-index.html)                                     | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-5c6f80cea6f85badc7483f45b30ac1e9fbbe87f5f14ab32f189f76817a7bbd58)     | New doc organization/navigation (general comments), SQL diagram                                                                                                                                  | [Link](https://www.cockroachlabs.com/docs/stable/alter-index.html)                            |
| [`ALTER RANGE`](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-range.html)                                     | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-ab7141ce39dd652cbf5b39f5c6a71db371cd7b48fd7c10096bc1f4f63a731f5d)     | New doc organization/navigation (general comments), SQL diagram                                                                                                                                  | [Link](https://www.cockroachlabs.com/docs/stable/alter-range.html)                            |
| [`ALTER FUNCTION`](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-function.html)                               | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-06cbeca723d3e128350762960ca565cbe23174ec96e037deefc87e0bbdf0d004)     | New doc organization/navigation (general comments), SQL diagram                                                                                                                                  | [Link](https://www.cockroachlabs.com/docs/stable/alter-function.html)                         |
| [`ALTER FUNCTION` examples](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-function.html#examples)             | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-06cbeca723d3e128350762960ca565cbe23174ec96e037deefc87e0bbdf0d004L43)  | New examples: Change owner, Change schema                                                                                                                                                        | [Link](https://www.cockroachlabs.com/docs/stable/alter-function.html#examples)                |
| [`DROP SECONDARY REGION`](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-database.html#drop-secondary-region)  | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-3f61c52b06ae762b3d60c38a7be6d3eee4c1b426ec98bab4989e021686ad17caR199) | "Drop secondary index" is now "unset secondary index" in the description                                                                                                                         | [Link](https://www.cockroachlabs.com/docs/stable/drop-secondary-region.html)                  |
| [Table storage parameters](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/alter-table.html#table-storage-parameters) | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-6985d70fb116407311be568c6fe4069e0d5d8cb76823689177c67cd2ae2a8aaa)     | Previously, the TTL storage parameters were missing in this section. Are all relevant parameters included here?                                                                                  | [Link](https://www.cockroachlabs.com/docs/stable/set-storage-parameter.html#table-parameters) |
| [SQL Audit Logging](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/v22.2/sql-audit-logging)                                 | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-a08e58ca544cd58dda3b538c11b66ef10f6479dc2ceae82789c4f548f36544a6)     | "File format", "File storage location", and "Performance considerations" were moved (content unchanged) to this doc. Previously, those sections were in the `EXPERIMENTAL_AUDIT` reference page. | [Link](https://www.cockroachlabs.com/docs/stable/experimental-audit.html)                     |
| [SQL Statements](https://deploy-preview-16151--cockroachdb-docs.netlify.app/docs/stable/sql-statements.html)                                 | [Link](https://github.com/cockroachdb/docs/pull/16151/files#diff-3fc12f363897873a4ed0887059a447a6082be00e8265035167ea5e882d1a6f09)     | The subcommands were removed from the tables.                                                                                                                                                    | [Link](https://www.cockroachlabs.com/docs/stable/sql-statements.html)                         |

The following issues were caused by separating SQL subcommand pages:

- Linking across subcommand and main statement pages (for example, the various multi-region commands for `ALTER DATABASE`), including in the `See also` section at the bottom of each page, requires users to jump between pages without seeing a centralized representation of the syntax. This update makes the statement hierarchy clear and tightens navigation between subcommands for a given statement.
- SQL subcommand pages have not been consistently updated when they apply to more than one statement. The new structure makes this less likely to happen, since a subcommand will now have to be explicitly added to the relevant page.
- Some subcommands were being named in docs using incorrect/nonexistent syntax, such as `RENAME DATABASE` for `ALTER DATABASE ... RENAME TO`. This is corrected where I encountered it, and should no longer happen.
- Likewise, the docs have been inconsistent in how they generally reference subcommands. The new structure encourages naming subcommands together with a parent statement, as this is usually appropriate.
- Docs pages inconsistently link to either the "guide" pages for a feature or to the dedicated reference pages for SQL subcommands. This restructure makes it clearer that the SQL reference pages and feature documentation serve different purposes for users.
- Some tutorials and guides were linking out to subcommand docs that apply to more than one statement (e.g., `PARTITION BY`) rather than to the exact subcommand used in a code snippet (e.g., `ALTER TABLE ... PARTITION BY`). Doing the latter is more helpful and immediate.

This PR has the following overall effect on the docs:

- The subcommands are located below the main statements, and this generally matches the PostgreSQL docs approach. It is easier to understand how the full SQL statements are structured.
- The updated docs have a single consistent structure and are therefore easier to update when new syntax is added for a statement.
- Parameter definitions for the various subcommands are more thoroughly, consistently, and correctly documented.
- Some SQL statement docs are much longer (particularly `ALTER TABLE`), as each statement's subcommands (and subcommand examples) are now contained on a single page. This makes updating each doc a little more intensive as the TW has to navigate more lines. However, for the aforementioned reasons, I think these docs are ultimately easier to maintain (and understand as a reader).
- `ALTER DATABASE`, `ALTER TABLE`, and `ALTER INDEX` pages now have SQL diagrams that display the full syntax. Previously, each of their subcommands had its own dedicated SQL diagram. This update trades specificity for simplicity and completeness (again, this generally matches the approach in the PostgreSQL docs).
- The nav bar for the SQL reference is simplified, since it only contains the main statements.

Full(er) summary of changes:

- Consolidated the subcommand pages for `ALTER DATABASE`, `ALTER TABLE`, and `ALTER INDEX` into the reference pages for the parent statements. This consists of:
  - Moving the high-level descriptions, considerations, required privileges, and parameter definitions for each subcommand underneath a subheading on the statement page.
  - Grouping subcommand examples under appropriate subheadings at the bottom of the statement page.
  - Linking directly from the subcommand subheadings to the corresponding example subheadings.
- Added references to new generated SQL diagrams for `ALTER DATABASE`, `ALTER TABLE`, and `ALTER INDEX` (see https://github.com/cockroachdb/cockroach/pull/95591 and https://github.com/cockroachdb/cockroach/pull/96030).
- Updated some docs and their corresponding SQL diagrams with clearer parameter names (e.g., when renaming a function, the corresponding diagram now refers to `function_new_name` instead of `name`).
- Made opportunistic fixes to SQL parameter documentation where references had been missing or diverged from the generated diagram.
- Added some new `ALTER FUNCTION` examples where coverage had been missing.
- Moderately refactored and consolidated the various multi-region SQL examples, which had a lot of overlap once brought into the same doc (`ALTER DATABASE`).
- Amended the description of `DROP SECONDARY REGION` from "drop" to "unset". I think this is more accurate, as it doesn't remove the region, only the "secondary region" designation.
- Amended `ALTER DATABASE ... PLACEMENT RESTRICTED` to `ALTER DATABASE ... PLACEMENT` as `RESTRICTED` is just one of two possible parameters for the subcommand.
- Moved the list of row-level TTL parameters into `table-storage-parameters.md` so that the TTL parameters appear with the relevant commands for setting storage parameters (they had been lost earlier).
- Moved some of the content on the `EXPERIMENTAL_AUDIT` subcommand page to the `sql-audit-logging.md` feature page, as it is a better fit there.
- Made opportunistic copyedits to fix various instances of the issues listed at the top of this description.
- Removed individual subcommand pages, added `301` redirects for them, and manually changed the links in v22.2 that I could find. In some cases, it seemed best to redirect links to feature pages rather than specific SQL reference pages (e.g., `configure-replication-zones.html` and `partitioning.html`).
- Removed the subcommands from the SQL statements page (`sql-statements.md`). Only the ones with separate pages were listed here, and some were named incorrectly (e.g. `RENAME DATABASE`).
- Removed subcommand pages from the side navigation.
- Updated the CSS so that code-formatted strings in subheadings are highlighted, as in body text. This helps with readability.

Follow-up work for a separate ticket:

- Some statements are already documented together with their subcommands, but not with the same structure that I've used in this PR. This includes `ALTER ROLE`, `ALTER SCHEMA`, `ALTER SEQUENCE`, `ALTER TYPE`, `ALTER VIEW`. These should be updated to match the format used in this PR (if reviewers find it acceptable).
- The new/updated SQL diagrams may have surfaced some syntax that still isn't documented, such as `IF NOT EXISTS` for certain statements. These should be covered as well.

nickvigilante commented on v22.2/reassign-owned.md:
```diff
@@ -10,7 +10,7 @@ The `REASSIGN OWNED` statement changes the [ownership](security-reference/author
 
 {{site.data.alerts.callout_success}}
-To change the ownership of any single object (e.g., a table or a database), use the [`OWNER TO`](owner-to.html) subcommand of the object's [`ALTER` statement](sql-statements.html).
+To change the ownership of any single object (e.g., a table or a database), use the `OWNER TO` subcommand of the object's [`ALTER` statement](sql-statements.html).
```
```suggestion
To change the ownership of any single object (e.g., a table or a database), use the [`OWNER TO`](alter-table.html#owner-to) subcommand of the object's [`ALTER` statement](sql-statements.html).
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -5,13 +5,13 @@ toc: true
 ---
 
-The `RENAME CONSTRAINT` [statement](sql-statements.html) changes the name of a constraint on a column.
+<!-- The `RENAME CONSTRAINT` [statement](sql-statements.html) changes the name of a constraint on a column. -->
```
```suggestion
{% comment %}The `RENAME CONSTRAINT` [statement](sql-statements.html) changes the name of a constraint on a column.{% endcomment %}
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -5,13 +5,13 @@ toc: true
 {% include {{ page.version.version }}/misc/schema-change-stmt-note.md %}
 
-{{site.data.alerts.callout_info}}
+<!-- {{site.data.alerts.callout_info}}
```
```suggestion
{% comment %}{{site.data.alerts.callout_info}}
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -5,13 +5,13 @@ toc: true
+<!-- {{site.data.alerts.callout_info}}
 It is not possible to rename a constraint for a column referenced by a view. For more details, see [View Dependencies](views.html#view-dependencies).
-{{site.data.alerts.end}}
+{{site.data.alerts.end}} -->
```
```suggestion
{{site.data.alerts.end}}{% endcomment %}
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -23,22 +23,22 @@ It is not possible to rename a constraint for a column referenced by a view. For
 
 ## Required privileges
```
If the contents of this section are all being commented out, you should comment out the header, too.

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -23,22 +23,22 @@ It is not possible to rename a constraint for a column referenced by a view. For
 ## Required privileges
 
-The user must have the `CREATE` [privilege](security-reference/authorization.html#managing-privileges) on the table.
+<!-- The user must have the `CREATE` [privilege](security-reference/authorization.html#managing-privileges) on the table. -->
```
```suggestion
{% comment %}The user must have the `CREATE` [privilege](security-reference/authorization.html#managing-privileges) on the table.{% endcomment %}
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -23,22 +23,22 @@ It is not possible to rename a constraint for a column referenced by a view. For
+<!-- The user must have the `CREATE` [privilege](security-reference/authorization.html#managing-privileges) on the table. -->
 
-## Parameters
+<!-- ## Parameters
```
```suggestion
{% comment %}## Parameters
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -23,22 +23,22 @@ It is not possible to rename a constraint for a column referenced by a view. For
  `table_name` | The name of the table with the constraint you want to rename.
  `current_name` | The current name of the constraint.
- `name` | The new [`name`](sql-grammar.html#name) you want to use for the constraint, which must be unique to its table and follow these [identifier rules](keywords-and-identifiers.html#identifiers).
+ `name` | The new [`name`](sql-grammar.html#name) you want to use for the constraint, which must be unique to its table and follow these [identifier rules](keywords-and-identifiers.html#identifiers). -->
```
```suggestion
 `name` | The new [`name`](sql-grammar.html#name) you want to use for the constraint, which must be unique to its table and follow these [identifier rules](keywords-and-identifiers.html#identifiers).{% endcomment %}
```

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -23,22 +23,22 @@ It is not possible to rename a constraint for a column referenced by a view. For
 {% include {{ page.version.version }}/misc/schema-change-view-job.md %}
 
-## Example
+<!-- ## Example
```
```suggestion
{% comment %}## Example
```

taroface replied:
This page is also removed now!

nickvigilante commented on v22.2/rename-constraint.md:
```diff
@@ -81,16 +81,16 @@ The user must have the `CREATE` [privilege](security-reference/authorization.htm
   logon      | unique_customer_id_sales_id | UNIQUE          | UNIQUE (customer_id ASC, sales_id ASC) |   true
 (2 rows)
-~~~
+~~~ -->
```
```suggestion
~~~{% endcomment %}
```

nickvigilante commented on v22.2/set-locality.md:
```diff
@@ -1,296 +0,0 @@
----
```
It looks like this file is empty and not fully deleted. Did you mean to delete this file?

taroface replied:
Yes! Thanks for the catch.

nickvigilante commented on v22.2/with-storage-parameter.md:
```diff
@@ -40,10 +40,6 @@ The `WITH (storage parameter)` [statement](sql-statements.html) sets a storage p
 
-### Row-level TTL parameters
-
-For the list of storage parameters that affect how [Row-Level TTL](row-level-ttl.html) works, see the list of [TTL storage parameters](row-level-ttl.html#ttl-storage-parameters).
```
Nit: Just curious about this removal. It seems to still point to valid info on that page.

taroface replied:
That block has been moved into the `table-storage-parameters.md` include just above, as (after discussing with Rich) I believe it's relevant everywhere that include is used.

nickvigilante commented on _includes/v22.2/sql/range-splits.md:
```diff
@@ -0,0 +1,7 @@
+
+However, there are reasons why you may want to perform manual splits on the ranges that store tables or indexes:
+
+- When a table only consists of a single range, all writes and reads to the table will be served by that range's [leaseholder](architecture/replication-layer.html#leases). If a table only holds a small amount of data but is serving a large amount of traffic, load distribution can become unbalanced. Splitting the table's ranges manually can allow the load on the table to be more evenly distributed ...
```
Should this be "table or index" instead of just "table"? I see this include is referenced in alter-index.md, but after reading this, I'm not sure if it should apply to indexes, too.

taroface replied:
Good catch - I'm not sure and will defer to the technical review on this question!

nickvigilante reviewed (changes requested):
Great work so far! You have some htmltest errors of which I'm sure you're aware. I left a few comments and nits, but my general feedback is as follows:

- Avoid HTML comments (`<!--`, `-->`) and use Liquid comments instead (`{% comment %}`, `{% endcomment %}`).
- The `_redirects` file needs to have `dev`, `stable`, and `v22.2` all labeled as redirects across the file. You can use a Liquid block to save some screen real estate, and I can help you craft that if you want.

taroface commented on _redirects:
```diff
@@ -381,6 +381,38 @@ layout: null
 /docs/cockroachcloud/backups-page.html /docs/cockroachcloud/use-managed-service-backups.html 301
 
 # Removed pages
+/docs/stable/add-column.html /docs/stable/alter-table.html#add-column 301
```
Added these.

taroface commented on v22.2/alter-range-relocate.md:
```diff
@@ -5,11 +5,11 @@ toc: true
 ---
 
-The `ALTER RANGE ... RELOCATE` statement is a subcommand of [`ALTER RANGE`](alter-range.html). It is used to move a lease or [replica](architecture/overview.html#architecture-replica) between [stores](cockroach-start.html#store). This is helpful in an emergency situation to relocate data in the cluster.
+<!-- The `ALTER RANGE ... RELOCATE` statement is a subcommand of [`ALTER RANGE`](alter-range.html). It is used to move a lease or [replica](architecture/overview.html#architecture-replica) between [stores](cockroach-start.html#store). This is helpful in an emergency situation to relocate data in the cluster.
```
My bad - had intended to remove this page. It's gone now.

nickvigilante reviewed (approved):
If the content was just being moved from the individual pages to the main alter-X pages, this LGTM!

taroface commented:
> If the content was just being moved from the individual pages to the main alter-X pages, this LGTM!

TFTR! There are other types of changes, including corrections and enhancements to the SQL docs, as mentioned in the description. But yes, reorganizing the existing docs content was the main purpose here.

rytaft commented on v22.2/alter-database.md:
```diff
@@ -1,34 +1,1082 @@
+
+Parameter | Description |
+----------|-------------|
+`region_name` | Usually, the [region](multiregion-overview.html#database-regions) being set as the secondary region for this database. E.g., `"ap-southeast-2"`. Allowed values include any (non-primary) region present in `SHOW REGIONS`. |
```
Why "Usually"?

rytaft commented on v22.2/alter-database.md:
```diff
@@ -1,34 +1,1082 @@
+      WHEN city = 'los angeles' THEN 'us-west1'
+ END
+) STORED;
+ALTER TABLE rides ALTER COLUMN REGION SET NOT NULL;
```
nit: REGION -> region

taroface replied:
Fixed.

rytaft commented on v22.2/alter-database.md:
```diff
@@ -1,34 +1,1082 @@
+
+## See also
+
+tk
```
tk?

taroface replied:
Hmm, this placeholder should have been updated and I don't see it anymore. So will mark as resolved!

rytaft reviewed (commented):
I read through most of the files before I realized that wasn't really want you intended. But I left a few comments on things that sounded a bit odd to me. I didn't review alter-table.md

taroface commented:
> I read through most of the files before I realized that wasn't really want you intended. But I left a few comments on things that sounded a bit odd to me. I didn't review alter-table.md

Sorry that I made this unclear. I've updated the PR with a table that points to exactly what needs review.

TFTRs on the content so far -- some of the lines you commented on are pre-existing content, and I will either address in a follow-up commit to this PR or file new issues for them.

rytaft commented:
Thanks so much for the table with the detailed reviewer info! Very helpful.

I agree that this organization is a big improvement. Some comments:
- On the main 5 pages, inside a given subcommand, you have a Parameters table at the bottom of the section. If possible, I think it would help to show the SQL diagram for just the given subcommand in that section so readers can see where/how the parameters are used. Right now, I need to scroll to the top of the page to see the giant SQL diagram and figure out where the parameters are used (the big diagram is good too, but I'd love to see both).
- On the main 5 pages, the location of View Schema Changes (after all the subcommands and before the examples) feels a bit odd. It's a tiny section and feels a bit out of place. (Not sure I have a better suggestion, so feel free to ignore). Also the section doesn't exist for ALTER RANGE -- I suppose ALTER RANGE commands are not considered schema changes / jobs?
- ALTER FUNCTION seems to be missing descriptions of some of the subcommands.
- There are several table-statistics related storage parameters that are missing. They are described [here](https://github.com/cockroachdb/docs/issues/13783) and [here](https://github.com/cockroachdb/docs/issues/14965).

taroface commented on v22.2/alter-database.md:
```diff
@@ -1,34 +1,1082 @@
 ---
 
-The `ALTER DATABASE` [statement](sql-statements.html) applies a schema change to a database. For information on using `ALTER DATABASE`, see the pages for its relevant [subcommands](#subcommands).
+The `ALTER DATABASE` [statement](sql-statements.html) applies a [schema change](online-schema-changes.html) to a database. Combine this statement with its [subcommands](#subcommands).
```
Removed in all docs.

taroface commented:
TFTR @rytaft! See responses below.

>     * On the main 5 pages, inside a given subcommand, you have a Parameters table at the bottom of the section. If possible, I think it would help to show the SQL diagram for just the given subcommand in that section so readers can see where/how the parameters are used. Right now, I need to scroll to the top of the page to see the giant SQL diagram and figure out where the parameters are used (the big diagram is good too, but I'd love to see both).

I've thought the same and will need to save this for a follow-up ticket (ideally to be handled sooner than later). I need to check that the existing diagrams for each subcommand are still correct, and create other ones that are missing, but it will take a bit of time.

>     * On the main 5 pages, the location of View Schema Changes (after all the subcommands and before the examples) feels a bit odd. It's a tiny section and feels a bit out of place. (Not sure I have a better suggestion, so feel free to ignore). Also the section doesn't exist for ALTER RANGE -- I suppose ALTER RANGE commands are not considered schema changes / jobs?

These seem extraneous to me now, so I've moved those contents into the intro sections on each page.

>     * ALTER FUNCTION seems to be missing descriptions of some of the subcommands.

Hmm, I don't see any missing descriptions. Am I overlooking them?

>     * There are several table-statistics related storage parameters that are missing. They are described [here](https://github.com/cockroachdb/docs/issues/13783) and [here](https://github.com/cockroachdb/docs/issues/14965).

Thanks for pointing this out! I added them to the table (and the v22.1 table, for the auto stats parameters).

rytaft commented:
The changes LGTM

For ALTER FUNCTION, I could be wrong, but looking at the SQL diagram, it just looks like there are more subcommands than `OWNER TO`, `RENAME TO` and `SET SCHEMA` (e.g., `CALLED`, `STRICT`, etc).

taroface commented:
> The changes LGTM
> 
> For ALTER FUNCTION, I could be wrong, but looking at the SQL diagram, it just looks like there are more subcommands than `OWNER TO`, `RENAME TO` and `SET SCHEMA` (e.g., `CALLED`, `STRICT`, etc).

Ah I see. I've added a link to the section of the User-Defined Functions doc that explains these. I'll file an issue in case they also need to be documented as parameters.

rytaft reviewed (approved):
LGTM thanks!

kathancox commented on v22.2/alter-table.md:
```diff
@@ -9,36 +9,2977 @@ docs_area: reference.sql
+
+The user must have the `CREATE` [privilege](security-reference/authorization.html#managing-privileges) on the table.
+
+#### Parameters
```
I wonder if instead of including another diagram for each subcommand (which you mentioned as a future possibility) you could add a sentence to each parameter section directing the reader to the synopsis diagram at the top? Each parameter is called out in the overarching diagram? (I think...)

taroface replied:
Really good simple suggestion - I've added this to all the updated docs!

kathancox reviewed (approved):
Hey Ryan! LGTM! So good. This is so much work, but I think the changes you've made here are a big improvement and consolidate information in an efficient way. None of the comments I've made are blocking, most are style issues that you might consider opening issues for (or not). I think there was like 1 typo that I spotted and smaller things you might change in this PR. Other things I was thinking about while reviewing that aren't things to change in the PR:
- You might want to create some kind of standard layout for SQL pages kind of based on this work. We do have this already, but not officially.
- I wonder if the subcommands removal from the navbar might trip some people up who are used to our navigation.... probably not? And I don't think there is really a solution, beyond the redirects that you have done.
- Thanks so much for the PR description to help reviewers navigate through the PR — that was awesome.
- All my other thoughts regarding the change in structure for these pages are included in the comments. But overall, I think this is a good optimization.
- A couple things I really liked: The lengthy example sections prefaced with the `demo` example to follow along. The consistency throughout in terms of directing users to the right places and how the subcommand sections were structured was spot on.

kathancox commented on v22.2/alter-database.md:
```diff
@@ -1,34 +1,1129 @@
+
+### `ADD SUPER REGION`
+
+{% include feature-phases/preview.md %}
```
I think this is actually a wider question of where the Enterprise license label should be placed. (So maybe not solvable in this PR)

taroface replied:
Agreed - I've moved the preview include in all relevant locations.

taroface commented on v22.2/alter-database.md:
```diff
@@ -1,34 +1,1129 @@
+ALTER DATABASE movr SET PRIMARY REGION "us-east1";
+~~~
+
+~~~
```
I'll file a Style issue if we don't already have one. I'd like to keep consistent for now, since the snippet shows what the successful output looks like. Arguably it's the output that should be changed.

taroface commented on v22.2/alter-table.md:
```diff
@@ -9,36 +9,2977 @@ docs_area: reference.sql
+(13 rows)
+~~~
+
+Next, issue the [`SHOW PARTITIONS`](show-partitions.html) statement. The output below (which is edited for length) will verify that the unique index was automatically [partitioned](partitioning.html) for you. It shows that the `user_email_unique` index is now partitioned by the database regions `europe-west1`, `us-east1`, and `us-west1`.
```
Phew, these were pretty bad. Fixed!

taroface commented on v22.2/alter-table.md:
```diff
@@ -9,36 +9,2977 @@ docs_area: reference.sql
+{% include_cached copy-clipboard.html %}
+~~~ sql
+> WITH x AS (SHOW COLUMNS FROM customer) SELECT column_name, data_type FROM x WHERE column_name='c_discount';
+~~~
```
I'll file a Style issue as we're highly inconsistent in our approach with code snippets in tutorials.

taroface commented on v22.2/alter-index.md:
```diff
@@ -5,64 +5,421 @@ toc: true
+
+### Rename indexes
+
+#### Rename an index
```
Unfortunately yeah, to be consistent with how these example headings are named, and to avoid breaking any pre-existing links to the specific named example (short of manually changing them all), I need to keep both h4 and h5 here. I'll think on ways to improve this, however.
